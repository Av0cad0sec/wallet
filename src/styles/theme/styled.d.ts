import 'styled-components'
import { ThemeType } from 'grommet'

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends ThemeType {}
}
