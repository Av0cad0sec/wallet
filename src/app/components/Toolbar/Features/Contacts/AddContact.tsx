import { useContext } from 'react'
import { useDispatch } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { Box } from 'grommet/es6/components/Box'
import { Tabs } from 'grommet/es6/components/Tabs'
import { Tab } from 'grommet/es6/components/Tab'
import { ResponsiveContext } from 'grommet/es6/contexts/ResponsiveContext'
import { contactsActions } from 'app/state/contacts'
import { Contact } from 'app/state/contacts/types'
import { ResponsiveLayer } from '../../../ResponsiveLayer'
import { ContactAccountForm } from './ContactAccountForm'
import { layerOverlayMinHeight } from './layer'

interface AddContactProps {
  setLayerVisibility: (isVisible: boolean) => void
}

export const AddContact = ({ setLayerVisibility }: AddContactProps) => {
  const { t } = useTranslation()
  const isMobile = useContext(ResponsiveContext) === 'small'
  const dispatch = useDispatch()
  const submitHandler = (contact: Contact) => dispatch(contactsActions.add(contact))

  return (
    <ResponsiveLayer
      onClickOutside={() => setLayerVisibility(false)}
      onEsc={() => setLayerVisibility(false)}
      animation="none"
      background="background-front"
      modal
      position="top"
      margin={isMobile ? 'none' : 'xlarge'}
    >
      <Box margin="medium" width={isMobile ? 'auto' : '700px'}>
        <Tabs alignControls="start">
          <Tab title={t('toolbar.contacts.add', 'Add Contact')} style={{ textTransform: 'capitalize' }}>
            <Box
              flex="grow"
              justify="center"
              height={{ min: isMobile ? 'auto' : layerOverlayMinHeight }}
              pad={{ vertical: 'medium' }}
            >
              <ContactAccountForm setLayerVisibility={setLayerVisibility} submitHandler={submitHandler} />
            </Box>
          </Tab>
        </Tabs>
      </Box>
    </ResponsiveLayer>
  )
}
