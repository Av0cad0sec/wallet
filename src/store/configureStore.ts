import { routerMiddleware } from 'connected-react-router'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { createInjectorsEnhancer } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import { createReducer, history } from './reducers'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.message,
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })
  const { run: runSaga } = sagaMiddleware

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = configureStore({
    reducer: createReducer(),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production' || process.env.PUBLIC_URL.length > 0,
    enhancers,
    preloadedState: state,
  })

  return store
}
