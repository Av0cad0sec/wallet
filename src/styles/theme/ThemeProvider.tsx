import { Grommet, ThemeType } from 'grommet'
import * as React from 'react'

import { useThemeSlice } from './slice'

const defaultGrommetTheme: ThemeType = {
  global: {
    font: {
      family: 'Rubik',
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice()

  // const theme = useSelector(selectTheme);
  return (
    // <OriginalThemeProvider theme={theme}>
    <Grommet full theme={defaultGrommetTheme}>
      {React.Children.only(props.children)}
    </Grommet>
    // </OriginalThemeProvider>
  )
}
