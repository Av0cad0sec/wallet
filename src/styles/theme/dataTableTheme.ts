import { IDataTableStyles, ITheme } from 'react-data-table-component'

export const dataTableTheme = {
  background: { default: 'false' },
  highlightOnHover: {
    default: '#88888833',
    text: 'false',
  },
  text: {
    primary: 'false',
    secondary: 'false',
    disabled: 'false',
  },
  sortFocus: {
    default: 'false',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>

export const dataTableCustomStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}
