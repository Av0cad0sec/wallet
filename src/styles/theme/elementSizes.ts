export const sidebarSmallSizeLogo = 30
export const sidebarMediumSizeLogo = 40
export const mobileToolbarZIndex = 3
export const mobileToolbarButtonsZIndex = mobileToolbarZIndex + 1
export const buildPreviewBannerZIndex = mobileToolbarZIndex - 1

export const mobileFooterNavigationHeight = '4rem'
