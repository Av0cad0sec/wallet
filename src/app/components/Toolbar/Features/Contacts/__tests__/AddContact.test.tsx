import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { AddContact } from '../AddContact'

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <AddContact setLayerVisibility={() => {}} />
      </ThemeProvider>
    </Provider>,
  )

describe('<AddContact  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.baseElement).toMatchSnapshot()
  })
})
