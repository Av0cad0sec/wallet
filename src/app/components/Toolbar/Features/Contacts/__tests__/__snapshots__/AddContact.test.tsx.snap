// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddContact  /> should match snapshot 1`] = `
.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: auto;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: visible;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
  border-bottom: solid 2px #0092f6;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 6px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: auto;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 24px;
  padding-bottom: 24px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
}

.c13 {
  font-size: 18px;
  line-height: 24px;
  color: #0092f6;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c10:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus > circle,
.c10:focus > ellipse,
.c10:focus > line,
.c10:focus > path,
.c10:focus > polygon,
.c10:focus > polyline,
.c10:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus::-moz-focus-inner {
  border: 0;
}

.c10:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible) > circle,.c10:focus:not(:focus-visible) > ellipse,
.c10:focus:not(:focus-visible) > line,.c10:focus:not(:focus-visible) > path,
.c10:focus:not(:focus-visible) > polygon,.c10:focus:not(:focus-visible) > polyline,
.c10:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c23 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c23:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c23:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c23:focus > circle,
.c23:focus > ellipse,
.c23:focus > line,
.c23:focus > path,
.c23:focus > polygon,
.c23:focus > polyline,
.c23:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c23:focus::-moz-focus-inner {
  border: 0;
}

.c23:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c23:focus:not(:focus-visible) > circle,.c23:focus:not(:focus-visible) > ellipse,
.c23:focus:not(:focus-visible) > line,.c23:focus:not(:focus-visible) > path,
.c23:focus:not(:focus-visible) > polygon,.c23:focus:not(:focus-visible) > polyline,
.c23:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c23:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c24 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #F8F8F8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c24:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c24:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c24:focus > circle,
.c24:focus > ellipse,
.c24:focus > line,
.c24:focus > path,
.c24:focus > polygon,
.c24:focus > polyline,
.c24:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c24:focus::-moz-focus-inner {
  border: 0;
}

.c24:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c24:focus:not(:focus-visible) > circle,.c24:focus:not(:focus-visible) > ellipse,
.c24:focus:not(:focus-visible) > line,.c24:focus:not(:focus-visible) > path,
.c24:focus:not(:focus-visible) > polygon,.c24:focus:not(:focus-visible) > polyline,
.c24:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c24:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c14 {
  min-height: 0;
}

.c12 {
  white-space: nowrap;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
}

.c12:hover {
  color: #000000;
}

.c12:focus {
  z-index: 1;
}

.c1 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: transparent;
  position: relative;
  z-index: 20;
  pointer-events: none;
  outline: none;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
}

.c2 {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  background-color: rgba(0,0,0,0.5);
  color: #F8F8F8;
  pointer-events: all;
  will-change: transform;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 48px;
  background-color: #FFFFFF;
  color: #16213E;
  outline: none;
  pointer-events: all;
  z-index: 20;
  position: absolute;
  max-height: calc(100% - 0px - 0px);
  max-width: calc(100% - 0px - 0px);
  border-radius: 4px;
  top: 0px;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
}

.c4 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

.c20 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c20::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c20::-moz-placeholder {
  color: #AAAAAA;
}

.c20:-ms-input-placeholder {
  color: #AAAAAA;
}

.c20::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c20::-moz-focus-inner {
  border: none;
  outline: none;
}

.c20:-moz-placeholder,
.c20::-moz-placeholder {
  opacity: 1;
}

.c19 {
  position: relative;
  width: 100%;
}

.c21 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
  width: 100%;
  -webkit-appearance: none;
  max-width: 100%;
}

.c21::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c21::-moz-placeholder {
  color: #AAAAAA;
}

.c21:-ms-input-placeholder {
  color: #AAAAAA;
}

.c21::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c21::-moz-focus-inner {
  border: none;
  outline: none;
}

.c21:-moz-placeholder,
.c21::-moz-placeholder {
  opacity: 1;
}

@media only screen and (max-width:768px) {
  .c6 {
    margin: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin-top: 2px;
    margin-bottom: 2px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    border-bottom: solid 2px #0092f6;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-bottom: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c17 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c22 {
    padding-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    position: fixed;
    width: 100%;
    height: 100%;
    min-height: 100vh;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    position: relative;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    position: relative;
    max-height: none;
    max-width: none;
    border-radius: 0;
    height: 100vh;
    width: 100vw;
  }
}

<body
  style=""
>
  <div>
    <div
      class="c0"
      style="min-height: 100vh;"
    />
  </div>
  <div>
    <div
      aria-hidden="false"
    >
      <div
        class="c1"
        tabindex="-1"
      >
        <div
          class="c2"
        />
        <div
          class="c3"
          data-g-portal-id="0"
          style="overflow-y: auto;"
        >
          <a
            aria-hidden="true"
            class="c4"
            tabindex="-1"
          />
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c7 "
              >
                <div
                  class="c8"
                  role="tablist"
                >
                  <div
                    class="c9 "
                  >
                    <button
                      aria-expanded="true"
                      aria-selected="true"
                      class="c10"
                      role="tab"
                      style="text-transform: capitalize;"
                      type="button"
                    >
                      <div
                        class="c11 c12"
                      >
                        <span
                          class="c13"
                        >
                          toolbar.contacts.add
                        </span>
                      </div>
                    </button>
                  </div>
                </div>
                <div
                  aria-label="toolbar.contacts.add Tab Contents"
                  class="c14"
                  role="tabpanel"
                >
                  <div
                    class="c15"
                  >
                    <form
                      style="display: flex; flex: 1; flex-direction: column;"
                    >
                      <div
                        class="c16"
                      >
                        <div
                          class="c17 "
                        >
                          <div
                            class="c18 "
                          >
                            <div
                              class="c19"
                            >
                              <input
                                autocomplete="off"
                                class="c20"
                                name="name"
                                placeholder="toolbar.contacts.name"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="c17 "
                        >
                          <div
                            class="c18 "
                          >
                            <textarea
                              class="c21"
                              name="address"
                              placeholder="toolbar.contacts.address"
                              rows="3"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="c22"
                      >
                        <button
                          class="c23"
                          type="button"
                        >
                          toolbar.contacts.cancel
                        </button>
                        <button
                          class="c24"
                          type="submit"
                        >
                          toolbar.contacts.save
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
